CHANGELOG.md

```diff
@@ -48,6 +48,7 @@ This project adheres to [Semantic Versioning](http://semver.org/).
  - #2147, Ignore `Content-Type` headers for `GET` requests when calling RPCs. Previously, `GET` without parameters, but with `Content-Type: text/plain` or `Content-Type: application/octet-stream` would fail with `404 Not Found`, even if a function without arguments was available.
  - #2155, Ignore `max-rows` on POST, PATCH, PUT and DELETE - @steve-chavez
  - #2239, Fix misleading disambiguation error where the content of the `relationship` key looks like valid syntax - @laurenceisla
+ - #2254, Fix inferring a foreign key column as a primary key column on views - @steve-chavez
```

Comment:
This PR is confusing. The commit message says it's a refactor, which should not change any behavior. Yet, here it says it fixes a bug at the same time. And then there is a regression as noted in #2356.

Overall a lot of code that changed and not easy to understand for me what's happening and where the regression comes from.

Reply:
Yeah, sorry about that. I first did the refactor and then I noticed there was a fix.